engine/runtime/src/main/scala/org/enso/interpreter/instrument/job/ProgramExecutionSupport.scala

```diff
@@ -351,27 +351,32 @@ object ProgramExecutionSupport {
               VisualizationResult.findExceptionMessage(panic),
               ErrorResolver.getStackTrace(panic).flatMap(_.expressionId)
             )
-        case withWarnings: WithWarnings =>
-          val warningsCount = withWarnings.getWarningsCount
-          val warning =
-            if (warningsCount == 1) {
-              val warnings = withWarnings.getWarningsArray(null)
-              Option(ctx.executionService.toDisplayString(warnings(0).getValue))
-            } else {
-              None
-            }
-          Api.ExpressionUpdate.Payload.Value(
-            Some(
-              Api.ExpressionUpdate.Payload.Value
-                .Warnings(
-                  warningsCount,
-                  warning,
-                  withWarnings.isLimitReached()
+        case _ =>
+          if (WarningsLibrary.getUncached().hasWarnings(value.getValue)) {
```

Comment:
minor: `getUncached()` should be used as `getUncached` - linter tells me a warning: "Java accessor method called with empty argument clause"

Reply:
My IDE is silent, but here you are: 36d114d2c4